### PR TITLE
perf: cache spks + txs to avoid n-plus-one in bdk

### PR DESCRIPTION
**Pull request by bodymindarts**

### CODE REVIEW

File: src/payjoin/entity.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum PayjoinEvent {
+    Initialized {
+        foreign_utxos: Vec<ForeignPayjoinInput>,
+    },
+}
+
+pub struct PayJoin {
+    pub id: PayjoinId,
+
```
Comment (thevaibhav-dixit) [inline src/payjoin/entity.rs:18]: was this file intended to be committed ?

**Review by thevaibhav-dixit — APPROVED** [review]

File: src/payjoin/entity.rs (hunk 1 above)
Comment (bodymindarts) [reply src/payjoin/entity.rs:18]: whoops no
thanks
